### PR TITLE
feat: allow passing client options

### PR DESCRIPTION
**Pull request by david-plugge**
## What kind of change does this PR introduce?

Allow passing of client options for nextjs and remix https://github.com/supabase/auth-helpers/issues/420

## What is the current behavior?

Client options can´t be passed.

## What is the new behavior?

client options can be passed:

```ts
import { createBrowserSupabaseClient } from '@supabase/auth-helpers-nextjs';

const supabaseClient = createBrowserSupabaseClient({
	options: {
		...
	}
});
```

The auth key is omitted as it is set internally by the helper

### CODE REVIEW

File: packages/sveltekit/src/createClient.ts [hunk 1]
```diff
@@ -11,7 +10,7 @@ import type { TypedSupabaseClient } from './types';
 export function createClient(
   supabaseUrl: string,
   supabaseKey: string,
-  options?: SupabaseClientOptionsWithoutAuth<App.Supabase['SchemaName']>,
```
Comment (thorwebdev) [inline packages/sveltekit/src/createClient.ts:14]: Actually, I'd say the sveltekit option implementation is exactly what we want to use across the board. Sorry if that wasn't clear.

Comment (david-plugge) [reply packages/sveltekit/src/createClient.ts:14]: It should be the same for all, the SupabaseClientOptionsWithoutAuth type was unnecessary as the type `SupabaseClientOptions` exported from the shared package omits the auth key
https://github.com/supabase/auth-helpers/blob/7ddaa6a1dbb9262ea15620d661b6d1691a23623f/packages/shared/src/types.ts#L9

Comment (thorwebdev) [reply packages/sveltekit/src/createClient.ts:14]: ah yes, perfect, I missed that it was coming from shared, sweet 👍

Comment (david-plugge) [reply packages/sveltekit/src/createClient.ts:14]: Maybe we should also rename the type to reduce confusion

Comment (thorwebdev) [reply packages/sveltekit/src/createClient.ts:14]: Yah, I think that might be good

**Review by thorwebdev — APPROVED** [review]
LGTM, thanks

**Comment by thorwebdev** [comment]
@david-plugge if this is ready, can you run changeset for it? https://github.com/supabase/auth-helpers/blob/main/development.md#preparing-a-release

**Pull request closed without merging**